### PR TITLE
File service feature

### PR DESCRIPTION
**Pull request by nbittich**

### CODE REVIEW

File: config/dispatcher/dispatcher.ex [hunk 1]
```diff
@@ -49,11 +37,50 @@ defmodule Dispatcher do
     Proxy.forward conn, path, "http://resource/measure-concepts/"
   end
 
+  ###############################################################
+  # files
+  ###############################################################
+  get "/files/:id/download", %{ accept: [:any], layer: :api} do
```
Comment (Windvis) [inline config/dispatcher/dispatcher.ex:43]: I'm not sure I understand why the config needs to be so specific here. Any reason why the example from the docs doesn't work?

```ex
  match "/files/*path" do
    Proxy.forward conn, path, "http://file/files/"
  end
  ```

Is it to restrict certain requests in the future or something?

File: docker-compose.yml [hunk 2]
```diff
@@ -21,22 +36,51 @@ services:
...
   cache:
     image: semtech/mu-cache:2.0.1
     links:
       - resource:backend
+    labels:
+      - "logging=true"
+    restart: always
+    logging: *default-logging
+  file:
+    image: semtech/mu-file-service:3.1.1
```
Comment (Windvis) [inline docker-compose.yml:72]: It seems that 3.1.2 is the latest version. Is there a reason why that one can't be used?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,18 +1,33 @@
 version: '3.4'
 
+x-logging:
```
Comment (Windvis) [inline docker-compose.yml:3]: From a git history / commit readability point of view, it would have been better to split this in a separate commit I think, since it seems unrelated to file service integration. At least that's what I would have done anyway :smile:.

**Review by Windvis — COMMENTED** [review]
Just a couple of questions to understand everything better :smile:

File: config/dispatcher/dispatcher.ex (hunk 1 above)
Comment (nbittich) [reply config/dispatcher/dispatcher.ex:43]: the reason is we use two kind of microservices to handle the upload. One for the upload / download (mu-file-service), one for the metadata (mu-cl-resources / ember data). It's basically how it was done in loket

File: docker-compose.yml (hunk 2 above)
Comment (nbittich) [reply docker-compose.yml:72]: using the same version as loket.

File: docker-compose.yml (hunk 3 above)
Comment (nbittich) [reply docker-compose.yml:3]: for small changes like this, I don't mind having a single commit. basically to keep consistent with was done in loket;

File: docker-compose.yml (hunk 2 above)
Comment (Windvis) [reply docker-compose.yml:72]: I would go for the latest since that seems to include bugfixes. I'm not sure why we need to keep this in sync with another project.

File: docker-compose.yml (hunk 3 above)
Comment (Windvis) [reply docker-compose.yml:3]: Agree to disagree I guess :smile:

File: docker-compose.yml (hunk 2 above)
Comment (nbittich) [reply docker-compose.yml:72]: okay updated to 3.1.2
